spyder_remote_services/services/spyder_kernels/patches.py

```diff
@@ -0,0 +1,122 @@
+from functools import wraps
+import json
+from types import MethodType
+from typing import Any, Callable, Dict, List, ParamSpec, Tuple, TypeVar
+import uuid
+
+from jupyter_client.manager import KernelManager
+from jupyter_core.utils import ensure_async
+from jupyter_server._tz import isoformat
+from jupyter_server.auth.decorator import authorized
+from jupyter_server.services.kernels.handlers import MainKernelHandler
+from jupyter_server.services.kernels.kernelmanager import (
+    AsyncMappingKernelManager,
+)
+from jupyter_server.utils import url_escape, url_path_join
+from tornado import web
+from tornado.routing import Router
+
+from spyder_remote_services.services.spyder_kernels.provisioner import SpyderKernelProvisioner
+
+try:
+    from jupyter_client.jsonutil import json_default
+except ImportError:
+    from jupyter_client.jsonutil import date_default as json_default
+
+
+T = TypeVar("T")
+P = ParamSpec("P")
+
+class SpyderMainKernelHandler(MainKernelHandler):
+    @web.authenticated
+    @authorized
+    async def post(self):
+        """Start a kernel."""
+        km = self.kernel_manager
+        model = self.get_json_body()
+        if model is None:
+            model = {"name": km.default_kernel_name}
+        else:
+            model.setdefault("name", km.default_kernel_name)
+
+        kernel_id = await ensure_async(
+            km.start_kernel(  # type:ignore[has-type]
+                kernel_name=model["name"], path=model.get("path"),
+                spyder_kernel=model.get("spyder_kernel", False)
+            )
+        )
+
+        model = await ensure_async(km.kernel_model(kernel_id))
+        location = url_path_join(self.base_url, "api", "kernels", url_escape(kernel_id))
+        self.set_header("Location", location)
+        self.set_status(201)
+        self.finish(json.dumps(model, default=json_default))
+
+
+def __kernel_model(self, kernel_id):
+    """Return a JSON-safe dict representing a kernel
```

Comment:
```suggestion
    """
    Return a JSON-safe dict representing a kernel
```

We prefer starting docstrings in a new line for multi-line docstrings.